source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -56,17 +75,21 @@ Http::FilterHeadersStatus CacheFilter::decodeHeaders(Http::RequestHeaderMap& hea
         *decoder_callbacks_, headers);
     return Http::FilterHeadersStatus::Continue;
   }
+
   if (!CacheabilityUtils::canServeRequestFromCache(headers)) {
     ENVOY_STREAM_LOG(debug, "CacheFilter::decodeHeaders ignoring uncacheable request: {}",
                      *decoder_callbacks_, headers);
     return Http::FilterHeadersStatus::Continue;
   }
+
   ASSERT(decoder_callbacks_);
 
+  initRouteConfig();
```

Comment:
I think if you make initRouteConfig thread-safe (ignoring other potential issues of sharing) you may be tempted to add a mutex, which could be contended in this data-plane context.

Consider initializing config related stuff in the filter's Config object instead.

Reply:
Route config can't be initialized in the Config object, because it requires the route() that isn't even available until decodeHeaders. It has to be initialized per-filter, usually during decodeHeaders. (See, e.g. the call to [initConfig](https://github.com/envoyproxy/envoy/blob/0ab9440ec3082800f72ef7feeec393644f1fb89a/source/extensions/filters/http/buffer/buffer_filter.cc#L57) in buffer_filter.)